packages/tools/src/tools/annotation/UltrasoundAnnotationTool/utils/generateConvexHullFromContour.ts

```diff
@@ -0,0 +1,164 @@
+import type { Types } from '@cornerstonejs/core';
+
+/**
+ * Compute the perpendicular distance from point p to the line segment [a→b].
+ *
+ * @param {Types.Point2} p - Point to measure distance from
+ * @param {Types.Point2} a - First endpoint of line segment
+ * @param {Types.Point2} b - Second endpoint of line segment
+ * @returns {number} Perpendicular distance from p to line segment [a→b]
+ * @private
+ */
+function _perpDist(p: Types.Point2, a: Types.Point2, b: Types.Point2) {
+  const dx = b[0] - a[0],
+    dy = b[1] - a[1];
+  if (dx === 0 && dy === 0) {
+    // a and b are the same point
+    return Math.hypot(p[0] - a[0], p[1] - a[1]);
+  }
+  // projection t of p onto ab, clamped [0,1]
+  const t = ((p[0] - a[0]) * dx + (p[1] - a[1]) * dy) / (dx * dx + dy * dy);
+  const tClamped = Math.max(0, Math.min(1, t));
+  const projX = a[0] + tClamped * dx;
+  const projY = a[1] + tClamped * dy;
+  return Math.hypot(p[0] - projX, p[1] - projY);
+}
+
+/**
+ * Ramer–Douglas–Peucker contour simplification.
+ * @param {Array<Types.Point2>} points  original contour
+ * @param {number} epsilon       max allowed deviation (pixels)
+ * @returns {Array<Types.Point2>}       simplified contour
+ */
+export function simplifyContour(
+  points: Array<Types.Point2>,
+  epsilon: number
+): Array<Types.Point2> {
+  if (points.length < 3) {
+    return points.slice();
+  }
+
+  let maxDist = 0,
+    index = 0;
+  const a = points[0],
+    b = points[points.length - 1];
+  // find point furthest from chord a→b
+  for (let i = 1; i < points.length - 1; i++) {
+    const d = _perpDist(points[i], a, b);
+    if (d > maxDist) {
+      maxDist = d;
+      index = i;
+    }
+  }
+  // if max deviation > ε, recurse, else approximate by [a,b]
+  if (maxDist > epsilon) {
+    const left = simplifyContour(points.slice(0, index + 1), epsilon);
+    const right = simplifyContour(points.slice(index), epsilon);
+    // concatenate, dropping duplicate at join
+    return left.slice(0, -1).concat(right);
+  } else {
+    return [a, b];
+  }
+}
+
+/**
+ * Moving-average smoothing of a closed or open contour
+ * @param {Array<Types.Point2>} points     original contour
+ * @param {number} windowSize       radius of smoothing window (integer)
+ * @returns {Array<Types.Point2>}          smoothed contour
+ */
+export function smoothContour(
```

Comment:
we already have packages/tools/src/utilities/planarFreehandROITool/smoothAnnotation.ts see if you can use that one